active/0003-extensible-resource-deployment-observability-improvements.md

```diff
@@ -0,0 +1,311 @@
+
+To address this issue, the deployments and deployment operations APIs must be updated to include additional information, such as:
+
+- Deployment resource ID
+- Bicep extension alias
+- Bicep extension name
+- Bicep extension version
+- Resource symbolic name
+- Resource type
+- Resource API version
+- Resource identifiers
+
+This additional data would enable users to identify a deployed extensible resource and map it to the corresponding resource definition in a Bicep file or an ARM template, providing users with comprehensive insights.
+
+## Detailed design
+
+### API change for `GET /Microsoft.Resources/deployments/{deploymentId}`
+
+#### Adding `extensions`
+
+The `Microsoft.Resources/deployments` API will be updated to include an `extensions` array in the GET response body. This modification is intended to incorporate information about extensions that are involved in executing the parent deployment and nested deployments.
+
+```diff
+{
+  ...
+  "type": "Microsoft.Resources/deployments",
+  "properties": {
+    ...
+    "providers": [
+      {
+        "namespace": "Microsoft.Storage",
+        "resourceTypes": [
+          {
+            "resourceType": "storageAccounts",
+            "locations": [
+              "eastus"
+            ]
+          }
+        ]
+      }
+    ],
++   "extensions": [
++     {
+        // deploymentId is used to uniquely identify an extension since a nested deployment could import the same extension with the same alias.
++       "deploymentId": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/my-resource-group/providers/Microsoft.Resources/deployments/my-deployment",
++       "alias": "az",
++       "name": "AzureResourceManager",
++       "version": "1.0.0"
++     },
++     {
++       "deploymentId": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/my-resource-group/providers/Microsoft.Resources/deployments/my-deployment",
++       "alias": "k8s",
++       "name": "Kubernetes",
++       "version": "1.27.8",
++       "config": {
++         // Out of the scope of this REP, but it is required to enable Deployment Stacks integration.  
++       }
++     }
+    ]
+  }
+}
+```
+
+#### Updating the `outputResources` to include more identifiable information
+
+To improve the current structure where each `outputResources` object only has a resource `id`, it's essential to add more properties for identifying extensible resources, which lack resource IDs. These additions should include the resource symbolic name, resource type, resource API version, resource identifier, the resource ID of the deployment containing the resource, and the Bicep extension alias.
+
+```diff
+{
+  ...
+  "type": "Microsoft.Resources/deployments",
+  "properties": {
+    "outputResources": [
```

Comment:
Do we need any changes to the `dependencies` property? I wasn't sure whether it's currently supported for extensible resources.

Reply:
Follow up: Need to determine if we need dependencies for the first pass or if a brute-force retry will be sufficient.

Reply:
I don't think we'll need changes for stacks extensibility support. I added a note about this in the unresolved questions. For deletion, we'd only want to know about container relationships (like that of an RG and a resource inside of it) and not data-only dependencies (such as using the output properties of some resource as input to another).